src/main/java/org/folio/circulation/domain/notice/schedule/ReminderFeeScheduledNoticeService.java

```diff
@@ -5,56 +5,75 @@
 
+import static org.folio.circulation.support.results.Result.ofAsync;
 import static org.folio.circulation.support.results.Result.succeeded;
 
 public class ReminderFeeScheduledNoticeService {
 
   protected static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
 
   private final ScheduledNoticesRepository scheduledNoticesRepository;
+  private final CalendarRepository calendarRepository;
 
-  public ReminderFeeScheduledNoticeService (ScheduledNoticesRepository scheduledNoticesRepository) {
-    this.scheduledNoticesRepository = scheduledNoticesRepository;
+
+  public ReminderFeeScheduledNoticeService (Clients clients) {
+    this.scheduledNoticesRepository = ScheduledNoticesRepository.using(clients);
+    this.calendarRepository = new CalendarRepository(clients);
   }
 
   public Result<LoanAndRelatedRecords> scheduleFirstReminder(LoanAndRelatedRecords records) {
     Loan loan = records.getLoan();
     if (loan.getOverdueFinePolicy().isReminderFeesPolicy()) {
-
-      OverdueFinePolicyRemindersPolicy.ReminderSequenceEntry firstReminder =
-        loan.getOverdueFinePolicy().getRemindersPolicy().getReminderSequenceEntry(1);
-
-      ScheduledNoticeConfig config =
-        new ScheduledNoticeConfig(
-          NoticeTiming.AFTER,
-          null, // recurrence handled using reminder fee policy
-          firstReminder.getNoticeTemplateId(),
-          firstReminder.getNoticeFormat(),
-          true);
-
-      ScheduledNotice scheduledNotice = new ScheduledNotice(
-        UUID.randomUUID().toString(),
-        loan.getId(),
-        null,
-        loan.getUserId(),
-        null,
-        null,
-        TriggeringEvent.DUE_DATE_WITH_REMINDER_FEE,
-        firstReminder.getPeriod().plusDate(loan.getDueDate()),
-        config
-      );
-
-      scheduledNoticesRepository.create(scheduledNotice);
+      ReminderConfig firstReminder =
+        loan.getOverdueFinePolicy().getRemindersPolicy().getFirstReminder();
+      instantiateFirstScheduledNotice(records, firstReminder)
+        .thenAccept(r -> r.after(scheduledNoticesRepository::create));
     } else {
       log.debug("The current item, barcode {}, is not subject to a reminder fees policy.", loan.getItem().getBarcode());
     }
     return succeeded(records);
   }
+
+  private CompletableFuture<Result<ScheduledNotice>> instantiateFirstScheduledNotice(
+    LoanAndRelatedRecords loanRecords,
+    ReminderConfig reminderConfig) {
+
+    final Loan loan = loanRecords.getLoan();
+
+    return reminderConfig.nextNoticeDueOn(
+        loan.getDueDate(), loanRecords.getTimeZone(), loan.getCheckoutServicePointId(), calendarRepository)
+      .thenCompose(nextDueTime ->
+        ofAsync(
+          new ScheduledNotice(
+            UUID.randomUUID().toString(),
+            loan.getId(),
+            null,
+            loan.getUserId(),
+            null,
+            null,
+            TriggeringEvent.DUE_DATE_WITH_REMINDER_FEE,
+            nextDueTime.value(),
```

Comment:
Again, calling .value() is kind of an anti-pattern with the Result class. What if nextDueTime is a failed result? Methods like after(), next() etc. should be used instead.